### PR TITLE
NICPS-13: Add logic to login to display Captcha module after 2 failed…

### PR DESCRIPTION
**Pull request by nileshpawar1**
… attempts of login

### CODE REVIEW

File: src/java/com/zimbra/cs/taglib/tag/LoginTag.java [hunk 1]
```diff
@@ -127,6 +135,10 @@ public void doTag() throws JspException, IOException {
             PageContext pageContext = (PageContext) jctxt;
             HttpServletRequest request = (HttpServletRequest) pageContext.getRequest();
 
+            if(request.getParameter("g-recaptcha-response")!= null && !isCaptchaValid("6LdfLE0UAAAAAC3rf4G4oepXp7H0AEEegdfeDvUA", request.getParameter("g-recaptcha-response"))){
```
Comment (plobbes) [inline src/java/com/zimbra/cs/taglib/tag/LoginTag.java]: If that is supposed to be a secret key, we don't want it hard coded.

File: src/java/com/zimbra/cs/taglib/tag/LoginTag.java [hunk 2]
```diff
@@ -266,4 +278,27 @@ public static void setTrustedCookie(HttpServletResponse response, String trusted
             ZimbraCookie.clearCookie(response, name);
         }
     }
+
+    public static boolean isCaptchaValid(String secretKey, String response) {
+        try {
+            String url = "https://www.google.com/recaptcha/api/siteverify?"
```
Comment (plobbes) [inline src/java/com/zimbra/cs/taglib/tag/LoginTag.java]: Both secretKey and response should be URL encoded.

Comment (nileshpawar1) [reply src/java/com/zimbra/cs/taglib/tag/LoginTag.java]: done

File: src/java/com/zimbra/cs/taglib/tag/LoginTag.java (hunk 1 above)
Comment (nileshpawar1) [reply src/java/com/zimbra/cs/taglib/tag/LoginTag.java]: This functionality is going to be covered in new separate ticket

**Pull request merged**